pkg/logql/log/parser.go

```diff
@@ -200,12 +201,16 @@ func unescapeJSONString(b []byte) string {
 		return ""
 	}
 	res := string(bU)
-	// rune error is rejected by Prometheus
-	for _, r := range res {
+
+	// rune error is rejected by Prometheus hence replacing them with space
+	removeInvalidUtf := func(r rune) rune {
```

Comment:
nit: please move this function outside of `unescapeJSONString` function to re-use it in logfmt parser ...